src/Integration.Vsix/SolutionAnalysisRequester.cs

```diff
@@ -0,0 +1,106 @@
+//-----------------------------------------------------------------------
+// <copyright file="SolutionAnalysisRequester.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Options;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class SolutionAnalysisRequester
```

Comment:
General questions - what to do about errors (exceptions) and case in which we don't find the things we want in reflection?